#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/factory/containers/protos/WoodenBox.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5773515607952077 0.5773515674429848 0.5773476793219721 2.0944038686710593
  position 6.080161012354172e-06 1.3172359762211917e-05 2.14543048225827
}
TexturedBackground {
}
TexturedBackgroundLight {
}
PointLight {
  attenuation 0 0 1
  intensity 21
  location 4 4 -1
  radius 10
  castShadows TRUE
}
RectangleArena {
}
DEF epuck E-puck {
  translation 0.375 -0.375 0
  rotation 0 0 1 1.5708000060964915
  controller "Lab6_task1"
  supervisor TRUE
  turretSlot [
    DEF Front DistanceSensor {
      translation 0.045 0 -0.011
      name "front distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Right DistanceSensor {
      translation 0.01 -0.035 -0.011
      rotation 0 0 1 -1.5707853071795865
      name "right distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Rear DistanceSensor {
      translation -0.025 0 -0.011
      rotation 0 0 1 3.1415
      name "rear distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Left DistanceSensor {
      translation 0.01 0.035 -0.011
      rotation 0 0 1 1.5708
      name "left distance sensor"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    InertialUnit {
      rotation -1 0 0 -5.307179586466759e-06
    }
    Camera {
      translation 0.043 0 -0.015
      name "front camera"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
    Camera {
      translation 0 0.043 -0.015
      rotation 0 0 1 1.57
      name "right camera"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
    Camera {
      translation -0.043 0 -0.015
      rotation 0 0 1 3.14
      name "rear camera"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
    Camera {
      translation 0 -0.043 -0.015
      rotation 0 0 1 -1.57
      name "left camera"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
WoodenBox {
  translation 0.25 0.01 0
  name "Maze Wall 1"
  size 0.012 0.5 0.2
}
WoodenBox {
  translation -0.25 0.25 0
  rotation 0 0 1 1.5707
  name "Maze Wall 2"
  size 0.012 0.5 0.2
}
WoodenBox {
  translation 0 -0.25 0
  rotation 0 0 1 1.5707
  name "Maze Wall 3"
  size 0.012 0.5 0.2
}
WoodenBox {
  translation 0 0.125 0
  name "Maze Wall 4"
  size 0.012 0.25 0.2
}
WoodenBox {
  translation -0.25 -0.125 0
  name "Maze Wall 5"
  size 0.012 0.25 0.2
}
WoodenBox {
  translation -0.25 0.125 0
  name "Maze Wall 5(1)"
  size 0.012 0.25 0.2
}
